pallets/xyk/src/lib.rs

```diff
@@ -2852,6 +2858,13 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 			ExistenceRequirement::KeepAlive,
 		)?;
 
+		let liquidity_assets_minted = multiply_by_rational_with_rounding(
+			first_asset_amount,
+			total_liquidity_assets,
+			first_asset_reserve,
+			Rounding::Down,
+		).ok_or(Error::<T>::UnexpectedFailure)?;
```

Comment:
it seems that in case of total_liquidity_issuance == 0 it will evalute to 0 :thinking: ?